### PR TITLE
add support for uploading via shiny new runalyze API

### LINKED ISSUE #638: Activities no longer upload to Runalyze.com
It seems like some recent changes to runalyze have broken the upload feature of this app. The 'https://runalyze.com/activity/add' page does not seem to exist anymore.

Here's the error in logcat when I try to upload an activity:

```
01-28 21:26:56.768 11609 14572 E Runalyze: Error importing into Runalyze: 
01-28 21:26:56.768 11609 14572 E Runalyze: java.io.FileNotFoundException: https://runalyze.com/activity/add
01-28 21:26:56.768 11609 14572 E Runalyze:      at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getInputStream(HttpURLConnectionImpl.java:251)
01-28 21:26:56.768 11609 14572 E Runalyze:      at com.android.okhttp.internal.huc.DelegatingHttpsURLConnection.getInputStream(DelegatingHttpsURLConnection.java:210)
01-28 21:26:56.768 11609 14572 E Runalyze:      at com.android.okhttp.internal.huc.HttpsURLConnectionImpl.getInputStream(Unknown Source:0)
01-28 21:26:56.768 11609 14572 E Runalyze:      at org.runnerup.export.RunalyzeSynchronizer.upload(RunalyzeSynchronizer.java:524)
01-28 21:26:56.768 11609 14572 E Runalyze:      at org.runnerup.export.SyncManager$11.doInBackground(SyncManager.java:644)
01-28 21:26:56.768 11609 14572 E Runalyze:      at org.runnerup.export.SyncManager$11.doInBackground(SyncManager.java:639)
01-28 21:26:56.768 11609 14572 E Runalyze:      at android.os.AsyncTask$2.call(AsyncTask.java:333)
01-28 21:26:56.768 11609 14572 E Runalyze:      at java.util.concurrent.FutureTask.run(FutureTask.java:266)
01-28 21:26:56.768 11609 14572 E Runalyze:      at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:245)
01-28 21:26:56.768 11609 14572 E Runalyze:      at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
01-28 21:26:56.768 11609 14572 E Runalyze:      at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
01-28 21:26:56.768 11609 14572 E Runalyze:      at java.lang.Thread.run(Thread.java:764)

```

### PR DESCRIPTION
**Pull request by craftyguy**
Fixes #638

### CODE REVIEW

File: app/src/org/runnerup/export/RunalyzeSynchronizer.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2016 rickyepoderi@yahoo.es
+ * Copyright (C) 2018 Clayton Craft <clayton@craftyguy.net>
```
Comment (gerhardol) [inline app/src/org/runnerup/export/RunalyzeSynchronizer.java]: Is this a rewrite or was it based on the previous version? If so both copyrights should be kept
It look like it is a clone of Strava though

Comment (craftyguy) [reply app/src/org/runnerup/export/RunalyzeSynchronizer.java]: It's a rewrite, not based on the previous RunalyzeSynchronizer. I did re-use parts of the Strava exporter, so I'll change the copyright to reflect that.

**Comment by mipapo** [comment]
@gerhardol You wrote a mail to me "key is stored in the repo though" -> You don't store API keys in a public repository? You don't even do this for strava?

**Comment by craftyguy** [comment]
Yea I'm actually not sure what @jonasoreland does for that. The client ID/keys are all null for all exporters that use oauth2... so maybe he injects the client ID/keys at build time on his system before uploading to the play store? Presumably the f-droid folks also need to request their own client ID/keys too?

**Comment by mipapo** [comment]
@craftyguy  @gerhardol Where do I find the strava keys in the repo? Exposing the private client secret to the public is a bad practice and breaks the security of the oauth implementation.

**Comment by gerhardol** [comment]
> Where do I find the strava keys in the repo? Exposing the private client secret to the public is a bad practice and breaks the security of the oauth implementation.

app/assets/
I know it is a bad practice but it is also good if more than the maintainer can test. 

If not setup by @jonasoreland the app can be forked, but a new app on Play is needed too. I do not know if the name and logo must be changed too if so.

**Comment by mipapo** [comment]
@gerhardol  Well at least they are crypted - Who can decrypt them? 
I don't think anyone need for testing purposes production api keys. 
There are two options how to test an API: Knowing the possible responses and fake the API call (not really make a network api call) in the test or having a Testing Api against one can test.

**Comment by gerhardol** [comment]
Closing in favor of #689

**Pull request closed without merging**